File: iosApp/iosApp/Utils/Strings.swift
import Foundation

enum Strings {
    // Common
    static let save = NSLocalizedString("save", comment: "")
    static let discard = NSLocalizedString("discard", comment: "")
    static let cancel = NSLocalizedString("cancel", comment: "")
    static let resume = NSLocalizedString("resume", comment: "")
    static let quit = NSLocalizedString("quit", comment: "")
    static let retry = NSLocalizedString("retry", comment: "")
    static let ok = NSLocalizedString("ok", comment: "")
    static let errorTitle = NSLocalizedString("error_title", comment: "")
    static let loading = NSLocalizedString("loading", comment: "")
    static let done = NSLocalizedString("done", comment: "")
    static let game = NSLocalizedString("game", comment: "")
    static let audio = NSLocalizedString("audio", comment: "")
    static let music = NSLocalizedString("music", comment: "")
    static let sfx = NSLocalizedString("sfx", comment: "")
    static let complication = NSLocalizedString("complication", comment: "")
    static let glance = NSLocalizedString("glance", comment: "")
    static let refresh = NSLocalizedString("refresh", comment: "")
    static let filter = NSLocalizedString("filter", comment: "")

    // Home Screen
    static let appTitle = NSLocalizedString("app_title", comment: "")
    static let startNewGame = NSLocalizedString("start_new_game", comment: "")
    static let resumeGame = NSLocalizedString("resume_game", comment: "")
    static let difficulty = NSLocalizedString("difficulty", comment: "")
    static let keyboardHint = NSLocalizedString("keyboard_hint", comment: "")
    static let newGame = NSLocalizedString("new_game", comment: "")

    // Game Screen
    static let next = NSLocalizedString("next", comment: "")
    static let score = NSLocalizedString("score", comment: "")
    static let lines = NSLocalizedString("lines", comment: "")
    static let time = NSLocalizedString("time", comment: "")
    static let gameOver = NSLocalizedString("game_over", comment: "")
    static let backToHome = NSLocalizedString("back_to_home", comment: "")
    static let gamePaused = NSLocalizedString("game_paused", comment: "")
    static let paused = NSLocalizedString("paused", comment: "")

    // Settings Screen
    static let settings = NSLocalizedString("settings_volume", comment: "")
    static let musicVolume = NSLocalizedString("music_volume", comment: "")
    static let sfxVolume = NSLocalizedString("sfx_volume", comment: "")
    static let visualTheme = NSLocalizedString("visual_theme", comment: "")
    static let pieceStyle = NSLocalizedString("piece_style", comment: "")
    static let musicTheme = NSLocalizedString("music_theme", comment: "")

    // History Screen
    static let history = NSLocalizedString("history_value", comment: "")
    static let noGamesYet = NSLocalizedString("no_games_yet", comment: "")
    static let startGamePrompt = NSLocalizedString("start_game_prompt", comment: "")
    static let loadingHistory = NSLocalizedString("loading_history", comment: "")
    static let historyTitle = NSLocalizedString("history_title", comment: "")

    // Formatted strings
    static func finalScore(_ score: Int) -> String {
        String(format: NSLocalizedString("final_score", comment: ""), score)
    }

    static func linesCleared(_ lines: Int) -> String {
        String(format: NSLocalizedString("lines_cleared", comment: ""), lines)
    }

    static func scoreLabel(_ score: Int) -> String {
        String(format: NSLocalizedString("score_label", comment: ""), score)
    }

    static func linesLabel(_ lines: Int) -> String {
        String(format: NSLocalizedString("lines_label", comment: ""), lines)
    }

    static func difficultyLabel(_ difficulty: String) -> String {
        String(format: NSLocalizedString("difficulty_label", comment: ""), difficulty)
    }
}

File: webApp/src/jsMain/kotlin/com/yet/tetris/ui/strings/Strings.kt
package com.yet.tetris.ui.strings

/**
 * Localized strings for the web application
 * Based on composeApp/src/commonMain/composeResources/values/strings.xml
 *
 * TODO: Add support for multiple languages
 */
object Strings {
    // Common
    const val SAVE = "Save"
    const val DISCARD = "Discard"
    const val CANCEL = "Cancel"
    const val RESUME = "Resume"
    const val QUIT = "Quit"
    const val RETRY = "Retry"
    const val OK = "Ok"
    const val ERROR_TITLE = "An Error Occurred"

    // Home Screen
    const val APP_TITLE = "Tetris Lite"
    const val START_NEW_GAME = "Start New Game"
    const val RESUME_GAME = "Resume Game"
    const val DIFFICULTY = "Difficulty"

    // Game Screen
    const val NEXT = "Next"
    const val SCORE = "Score"
    const val LINES = "Lines"
    const val TIME = "Time"
    const val GAME_OVER = "Game Over"
    const val BACK_TO_HOME = "Back to Home"
    const val GAME_PAUSED = "Game Paused"
    const val PAUSE_MESSAGE = "What would you like to do?"

    // Settings Screen
    const val GAME_SETTINGS = "Game Settings"
    const val VISUAL_THEME = "Visual Theme"
    const val PIECE_STYLE = "Piece Style"
    const val KEYBOARD_LAYOUT = "Keyboard Layout"
    const val SWIPE_LAYOUT = "Swipe Layout"
    const val MUSIC = "Music"
    const val MUSIC_THEME = "Music Theme"
    const val SOUND_EFFECTS = "Sound Effects"
    const val MUSIC_VOLUME = "Music Volume"
    const val SFX_VOLUME = "SFX Volume"
    const val AUDIO = "Audio"

    // History Screen
    const val GAME_HISTORY = "Game History"
    const val NO_GAMES_YET = "No games played yet"
    const val START_GAME_PROMPT = "Start a new game to see your history"

    // Formatted strings
    fun finalScore(score: Long) = "Final Score: $score"

    fun linesCleared(lines: Long) = "Lines Cleared: $lines"

    fun scoreLabel(score: Long) = "Score: $score"

    fun linesLabel(lines: Long) = "Lines: $lines"

    fun difficultyLabel(difficulty: String) = "Difficulty: $difficulty"
}
